libs/librepcb/project/schematics/graphicsitems/sgi_symbol.cpp

```diff
@@ -77,10 +79,40 @@ void SGI_Symbol::updateCacheAndRepaint() noexcept
 
     // polygons
     for (const Polygon& polygon : mLibSymbol.getPolygons()) {
+        // query polygon path and line width
         QPainterPath polygonPath = polygon.getPath().toQPainterPathPx();
         qreal w = polygon.getLineWidth()->toPx() / 2;
+
+        // update bounding rectangle
         mBoundingRect = mBoundingRect.united(polygonPath.boundingRect().adjusted(-w, -w, w, w));
-        if (polygon.isGrabArea()) mShape = mShape.united(polygonPath);
+
+        // update shape
+        if (polygon.isGrabArea()) {
+            QPainterPathStroker stroker;
+            stroker.setCapStyle(Qt::RoundCap);
+            stroker.setJoinStyle(Qt::RoundJoin);
+            stroker.setWidth(2 * w);
+            mShape = stroker.createStroke(polygonPath);
+        }
+    }
+
+    // circles
+    for (const Circle& circle : mLibSymbol.getCircles()) {
+        // get circle radius, including compensation for the stroke width
+        qreal w = circle.getLineWidth()->toPx() / 2;
+        qreal r = circle.getDiameter()->toPx() / 2 + w;
+
+        // get the bounding rectangle for the circle
+        QPointF center = circle.getCenter().toPxQPointF();
+        QRectF boundingRect = QRectF(QPointF(center.x() - r, center.y() - r), QSizeF(r, r));
```

Comment:
I guess the calculation of the bounding rect is wrong. The second argument of `QRectF` is called *size* (i.e. width/height), but you pass the radius instead of the diameter.

Reply:
Oh, well spotted. I think you're right :slightly_smiling_face: How could I have noticed that while testing? What's the effect of the bounding box on the behavior?

Reply:
It should be visible when enabling the graphics layer `GraphicsLayer::sDebugGraphicsItemsBoundingRects`, but unfortunately currently there is no way to enable it easily, so it needs to be done with an ugly hack in the source code ;)

The effect of a wrong bounding box is that the whole symbol disappears from the screen if the bounding box is outside the visible area. Mostly this happens of you scroll around, so that half of a symbol is outside the visible area, and the other half also disappears even if still in the visible area.

Reply:
Ah, thanks, good to know!